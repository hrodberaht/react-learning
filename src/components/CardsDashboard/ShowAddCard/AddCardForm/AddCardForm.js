import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { addCard } from '../../../../actions';

export class AddCardForm extends Component {
  state = {
    name: '',
    email: '',
  };

  handleChange = (e) => {
    this.setState({ [e.target.id]: e.target.value });
  };

  resetState = () => {
    this.setState({
      name: '',
      email: '',
    });
  }

  handleAdd = (e) => {
    e.preventDefault();
    const { name, email } = this.state;
    const { handleAddCard } = this.props;
    handleAddCard(name, email);
    this.resetState();
  };

  render() {
    const { name, email } = this.state;
    return (

      <div className="AddCard">
        <form onSubmit={this.handleAdd}>
          <input
            id="name"
            type="text"
            value={name}
            onChange={this.handleChange}
          />
          Email:
          <input
            id="email"
            type="email"
            value={email}
            onChange={this.handleChange}
          />
          <button
            id="submit"
            type="submit"
            disabled={name.length < 3}
          >
          Add
          </button>
        </form>
      </div>
    );
  }
}

export default connect(
  null,
  { handleAddCard: addCard },
)(AddCardForm);

AddCardForm.propTypes = {
  handleAddCard: PropTypes.func.isRequired,
};
