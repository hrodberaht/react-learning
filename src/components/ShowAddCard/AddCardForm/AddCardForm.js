import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { addCard } from '../../../actions';

class AddCardForm extends Component {
  state = {
    name: '',
    email: '',
  };

  handleChange = (e) => {
    this.setState({ [e.target.id]: e.target.value });
  };

  handleAdd = (e) => {
    e.preventDefault();
    const { name, email } = this.state;
    const { handleAddCard } = this.props;
    handleAddCard(name, email);
    this.setState({
      name: '',
      email: '',
    });
  };

  render() {
    const { name, email } = this.state;
    return (

      <div className="AddCard">
        <form onSubmit={this.handleAdd}>
          <input
            id="name"
            type="text"
            value={name}
            onChange={this.handleChange}
          />
          Email:
          <input
            id="email"
            type="email"
            value={email}
            onChange={this.handleChange}
          />
          <button
            id="submit"
            type="submit"
            disabled={name.length < 3}
          >
          Add
          </button>
        </form>
      </div>
    );
  }
}

AddCardForm.propTypes = {
  handleAddCard: PropTypes.func.isRequired,
};

export { AddCardForm };
export const ConnectedAddCardForm = connect(
  null,
  { handleAddCard: addCard },
)(AddCardForm);
