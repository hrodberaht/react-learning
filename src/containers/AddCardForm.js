import { connect } from 'react-redux';
import { addCard } from '../actions';
import AddCardForm from '../components/ShowAddCard/AddCardForm/AddCardForm';


const mapDispatchToProps = dispatch => ({
  handleAddCard: card => dispatch(addCard(card.name, card.email)),
});


export default connect(null, mapDispatchToProps)(AddCardForm);
