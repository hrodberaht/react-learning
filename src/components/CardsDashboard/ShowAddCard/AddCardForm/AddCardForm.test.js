import React from 'react';
import { shallow } from 'enzyme';
import { AddCardForm } from './AddCardForm';

describe('<ShowAddCard />', () => {
  let wrapper;
  const props = {
    isAddCardVisible: false,
    handleAddCard: jest.fn(),
  };

  const setValuesToInputs = () => {
    wrapper.find('#name').simulate('change', { target: { value: 'John', id: 'name' } });
    wrapper.find('#email').simulate('change', { target: { value: 'test@test.pl', id: 'email' } });
  };

  beforeEach(() => {
    wrapper = shallow(<AddCardForm {...props} />);
  });

  it('should render without crash', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call function handleChange', () => {
    wrapper.instance().handleChange = jest.fn();
    wrapper.update();
    setValuesToInputs();
    expect(wrapper.instance().handleChange).toHaveBeenCalled();
  });

  it('should change state after change input', () => {
    setValuesToInputs();
    expect(wrapper.state()).toEqual({ name: 'John', email: 'test@test.pl' });
  });

  it('should call handleAddCard function with name and email', () => {
    setValuesToInputs();
    wrapper.find('form').simulate('submit', { preventDefault() {} });

    expect(props.handleAddCard).toHaveBeenCalledWith('John', 'test@test.pl');
    expect(wrapper.state()).toEqual({ name: '', email: '' });
  });
});
