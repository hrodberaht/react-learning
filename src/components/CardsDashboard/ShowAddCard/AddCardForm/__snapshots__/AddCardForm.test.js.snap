// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShowAddCard /> should render without crash 1`] = `
<div
  className="AddCard"
>
  <form
    onSubmit={[Function]}
  >
    <input
      id="name"
      onChange={[Function]}
      type="text"
      value=""
    />
    Email:
    <input
      id="email"
      onChange={[Function]}
      type="email"
      value=""
    />
    <button
      disabled={true}
      id="submit"
      type="submit"
    >
      Add
    </button>
  </form>
</div>
`;
